x-pack/plugins/enterprise_search/server/lib/connectors/add_connector.ts

```diff
@@ -80,16 +102,22 @@ export const addConnector = async (
     last_sync_status: null,
     last_synced: null,
     name: input.index_name.startsWith('search-') ? input.index_name.substring(7) : input.index_name,
+    pipeline: {
```

Comment:
suggestion:
```
pipeline: connectorsPipelineMeta ? {
      extract_binary_content: connectorsPipelineMeta?.default_extract_binary_content,
      name: connectorsPipelineMeta?.default_name,
      reduce_whitespace: connectorsPipelineMeta?.default_reduce_whitespace,
      run_ml_inference: connectorsPipelineMeta?.default_run_ml_inference,
    } : null,
```

Reply:
Happy to make the change. For my edification, can you explain why? Is this related to your above suggestion to remove the ` | null` from the individual fields, and to have this more clearly imply that either all the meta is there or it is all null?

Reply:
Yup. It's also to prevent creating an empty pipeline object, because a lot of usage checks elsewhere would be of the shape 'does this object exist', which won't tell us much if we always create it.